lib/commands/touch.js

```diff
@@ -1,9 +1,51 @@
 const commands = {};
-
 //This is needed to make clicks on -image elements work properly
 commands.performActions = async function (actions) {
-  return await this.winAppDriver.sendCommand('/actions', 'POST', {actions});
+  switch (actions[0].id) {
+    case 'default keyboard':
+      this.log.info('Mapping keyboard actions');
+      return await performKeyboardCommands(this.winAppDriver, actions[0].actions);
```

Comment:
it seems redundant to me to pass properties of `this` as arguments. You could simply bind the function while calling it and use the same `this` context as the caller one.

Reply:
I'll look into binding a method, I'm not that familiar with js so I'm not sure what it entails

Reply:
example:

```js
return await performKeyboardCommands.bind(this)(...args);
```

then inside of performKeyboardCommands function you may call `this.winAppDriver` and `this` would be pointing to the same instance as in the caller method.

Read https://www.geeksforgeeks.org/javascript-function-binding/ for more details